### PR TITLE
Extension disclaimers

### LINKED ISSUE #4820: Manage user expectations around third-party extensions and the extension manager
From the dev meeting 18 Jul 2018:

- [ ] See if we can easily include a warning in the comment above the setting without the wrapping looking horrible.
- [ ] Pop up a dialog box when the extension setting switches from false to true #4925
- [ ] Distinguish jupyter packages (color? badge? something else?) #4925
- ~~Have a small warning in the extension search interface~~ The branding in the previous item seems sufficient, in connection with the dialog that pops up when you enable the manager.

---
Original text:

We should have a huge red bold disclaimer at the top of the extension manager results that says that these are third party packages with no guarantees and you install at your own risk!

Or we need to curate a very carefully chosen set of "approved" extensions (such as the ipywidgets one?), with a button to search for additional third party ones or something? Or just put a URL field that people can paste a package name into.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Fixes #4820.
1. Adds a badge to packages published under the `@jupyterlab` or `@jupyter-widgets` orgs.
1. Sorts packages from those orgs to the top of the list.
1. Gives a warning when you enable the extension manager that there are potential security risks.
1. Makes the styling of the extension manager a bit more consistent with the rest of the app.

Before:
image

After:
image

Dialog:
image

A few questions:
1. Am I missing any obvious npm orgs?
1. Is this too heavy-handed in promoting `@jupyterlab` org extensions?
2. How do we like the warning dialog language?

### CODE REVIEW

File: packages/extensionmanager-extension/src/index.ts [hunk 1]
```diff
@@ -101,3 +114,33 @@ function addCommands(app: JupyterLab, view: ExtensionView): void {
...
+    return showDialog({
+      title: 'Enable Extension Manager?',
+      body:
+        "Thanks for trying out JupyterLab's extension manager. " +
+        'The JupyterLab development team is excited to have a robust ' +
+        'third-party extension community. ' +
+        'However, we cannot vouch for every extension, ' +
+        'and some may introduce security risks. ' +
+        'Do you want to continue?',
+      buttons: [Dialog.cancelButton(), Dialog.okButton()]
```
Comment (blink1073) [inline packages/extensionmanager-extension/src/index.ts]: I think this should be a warnButton

**Comment by blink1073** [comment]
I'm fine with all three as is

**Comment by vidartf** [comment]
What was the rationale for removing the warning colorbar? I'm not against it, just curious.

**Comment by jasongrout** [comment]
What does the warning colorbar signify?

**Comment by jasongrout** [comment]
> What does the warning colorbar signify?

Ah, I see. It indicates the status from the server call: https://github.com/jupyterlab/jupyterlab/blob/master/jupyterlab/extension_manager_handler.py#L92

**Comment by ian-r-rose** [comment]
My rationale for removing the warning bar (summarized in 52770eb) was something like:
1. It is not very clear that it indicates a warning, nor does it give much indication of what that warning is. Indeed, I didn't know it was a warning bar until getting into the code base. I thought it was something like an "installed" indicator.
1. Relatedly, many extension installs produce some kind of warning. Some of those we should probably fix, but in practice the warning bar wasn't communicating much information to me.
1. There can also be warnings associated with the overall build, such as `WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).` Again, we may want to fix that, but it doesn't seem fair to assign it to an extension.
1. If a build produces a warning, it seems like it can come from any of the installed extensions (not just the one that you are installing at the moment). Actually associating the warning with a given extension seems fraught.

**Comment by jasongrout** [comment]
Can we make the two confirmation dialog buttons "Enable" and "Disable"? I think best practices is to make the buttons verbs, to make it very clear what is going to happen if you press it.

File: packages/extensionmanager-extension/src/index.ts [hunk 2]
```diff
@@ -101,3 +114,33 @@ function addCommands(app: JupyterLab, view: ExtensionView): void {
...
+  /**
+   * Show a warning dialog about extension security.
+   *
+   * @returns whether the user accepted the dialog.
+   */
+  export async function showWarning(): Promise<boolean> {
+    return showDialog({
+      title: 'Enable Extension Manager?',
+      body:
+        "Thanks for trying out JupyterLab's extension manager. " +
```
[the remark this answers is not in the document]
Comment (ian-r-rose) [reply packages/extensionmanager-extension/src/index.ts]: Can you explain more what you mean? As far as I know, leading and trailing spacing is ignored, but there is none of that here.

**Comment by ian-r-rose** [comment]
Sure, I have relabeled the buttons.

**Comment by jasongrout** [comment]
I haven't thoroughly reviewed the code, but I played with the end result, and looked at the code a bit. I'm +1 on merging, and hammering out other issues on another PR.

There are two other points in #4820 not addressed here: see https://github.com/jupyterlab/jupyterlab/issues/4820#issuecomment-406284194 for the discussion of those points.

**Comment by ian-r-rose** [comment]
I can take a look at the comment string here. I was under the impression that we decided *not* to add a permanent warning to the extension manager UI.

**Comment by blink1073** [comment]
Looks great, thanks!

**Pull request merged**